docs/proposals/metadata/external-host-tags.md

```diff
@@ -0,0 +1,86 @@
+# External Host Tags collection
+
+- Authors: Massimiliano Pippi
+- Date: 2018-01-15
+- Status: review
+- [Discussion](https://github.com/DataDog/datadog-agent/pull/1053)
+
+## Overview
+
+Some checks (at this moment there are only 2 of them: [vsphere][] and [openstack][])
+expose a method with the signature `get_external_host_tags()` that's called by
+the Agent whenever it's time to build and send the metadata payload. The way the
+new Agent collects metadata has drastically changed and this feature wasn't been
+ported yet, thus such checks are currently not working.
+
+## Problem
+
+Previous versions of the Agent took advantage of the dynamic nature of Python
+and invoked the collection method [like this][agent5-collection], an alternative
+must be defined to provide the new Agent with the same feature.
+
+## Constraints
+
+Changes to the checks must be compatible with both the agents 5.x and 6.x so that
+we don't need to backport the changes to Agent 5.
+
+## Recommended Solution
+
+### Change the collection model from "Pull" to "Push"
```

Comment:
We send metadata payload every 4 hours. I think the first metadata payload will be sent before the first run of any check from the collector. This means that host_tags from integration could take up to 4 hours to be displayed in the backend.
I think we should keep the "push" idea but trigger a new metadata collection every time a new host tag is added.

Reply:
Yep we can write a dedicated metadata provider with a meaningful interval for this, plus some logic to handle the cache, see also this comment: https://github.com/DataDog/datadog-agent/pull/1053#discussion_r161545984
Didn't want to include implementation details in the RFC but I think we're on the same page.

Reply:
👌, I'm fine to avoid implementation details, just wanted to make sure we were aware of it.